import "ads/bsyeti/big_rt/lib/processing/shard_processor/stateful/config/config.proto";
import "ads/bsyeti/big_rt/lib/consuming_system/config/config.proto";
import "ads/bsyeti/big_rt/lib/supplier/config/supplier_config.proto";

import "ads/bsyeti/libs/profiling/solomon/proto/config.proto";
import "ads/bsyeti/libs/ytex/http/proto/config.proto";
import "ads/bsyeti/libs/ytex/logging/proto/config.proto";

import "quality/user_sessions/rt/lib/common/protos/yt_client_config.proto";
import "quality/user_sessions/rt/lib/state_managers/proto/proto/config.proto";

package NAlice.NWonderlogs;

message TWonderlogsCreatorConfig {
    message TProcessorConfig {
        optional uint64 ShardsCount = 1;
        optional string OutputQueue = 2;
        optional string CompressionCodec = 3 [default = "null"];
        optional uint64 SleepMilliseconds = 4;
    }

    message TStateManagerConfigs {
        optional NUserSessions.NRT.TProtoStateManagerConfig UuidMessageId = 1;
        optional NUserSessions.NRT.TProtoStateManagerConfig UniproxyPrepared = 2;
        optional NUserSessions.NRT.TProtoStateManagerConfig MegamindPrepared = 3;
    }

    optional NBSYeti.NProfiling.TExporterConfig Exporter = 1;
    repeated NBigRT.TSupplierConfig Suppliers = 7;
    optional NBigRT.TConsumingSystemConfig ConsumingSystem = 8;
    optional NBigRT.TStatefulShardProcessorConfig StatefulShardProcessorConfig = 9;
    optional uint64 MaxInflightBytes = 11 [default = 1000000];
    optional uint32 SharedTransactionPeriod = 12;

    optional TStateManagerConfigs StateManagerConfigs = 16;

    optional NYTEx.NHttp.TServerConfig HttpServer = 18;
    optional NYTEx.NLogging.TConfig Logs = 19;
    optional TProcessorConfig ProcessorConfig = 20;

    optional NUserSessions.NRT.TYtClientConfig YtClientConfig = 21;
}
